doc/firewalld_services.md

```diff
@@ -0,0 +1,83 @@
+
+![summaryofchannges](https://user-images.githubusercontent.com/7056681/35337660-b5700d4c-0113-11e8-829f-0d92cdfe97e3.png)
+
+Being the implementation something like:
+
+```ruby
+
+Yast.import "CWMFirewallInterfaces"
+
+# You can still use "service:" prepend although it is recommended to remove
+# it when adapting the module
+settings = { "services" => ["service:cluster"], "display_details" => true }
+CWMFirewallInterfaces.CreateOpenFirewallWidget(settings)
+```
+
+In most of the cases the only requirement will be to define the service in `firewalld` and probabily it will be already
+provided by the `firewalld` package. In summary the changes in code should look like:
+
+```ruby
+
+# Require the new firewalld library and drop any import of SuSEFirewall2
+require 'y2firewall/firewalld'
+
+  # In your Module.Read method replace SuSEFirewalld.Read by firewalld.read
+    firewalld.read
+  # # In your Module.Write method replace SuSEFirewalld.Write by firewalld.read
+    firewalld.write
+  
+  private
+  
+    # Of course also define the firewalld convenience method to obtain the firewalld singlenton instance
+    def firewalld
+      Y2Firewall::Firewalld.instance
+    end
+```
+
+### Modify service ports definition
+
+The services definition that cames with firewalld can be modified. By default all the services that came with firewalld are placed in `/usr/lib/firewalld/services` although if a service is modified then it is placed in `/etc/firewalld/services` allowing the admin to go bak to the original definition is needed.
+
+SuSEFirewallServices was used in the past for modifying the services definition, it has been also dropped and the services configuration have to be done through the new class `Y2Firewall::Firewalld::Service`. 
+
+To modify the ports definition, the class method `modify_ports` has been provided making the call as similar to the old one (`SetNeededPortsAndProtocols`) as possible.
+
+```ruby
+      
+      # SuSEFirewallServices.SetNeededPortsAndProtocols("service:cluster", { "tcp_ports" => tcp_ports })
+      begin
+        Y2Firewall::Firewalld::Service.modify_ports(name: "cluster", tcp_ports: tcp_ports)
+      rescue Y2Firewall::Firewalld::Service::NotFound
+        y2error("Firewalld 'cluster' service is not available.")
+      end
+```
+
+To get the list of service ports we have to ask the service object itself.
+
+```ruby
+      # SuSEFirewallServices.GetNeededTCPPorts("service:cluster")
+      begin
+        fwd_cluster = firewalld.find_service("cluster")
+        tcp_ports = fwd_cluster.tcp_ports
+      rescue Y2Firewall::Firewalld::Service::NotFound
+        tcp_ports = []
+      end
+
+```
+
+## Important Note:
+
+If a `firewalld service` is not defined, then the `CWMFirewallInterfaces` widget will show a summary of not available services, and will suggest to deploy them to be able to configure the firewall.
```

Comment:
*summary of not available service* -> *a list of missing services*
*and will suggest* -> *suggesting*